libnmstate/nm/ipv6.py

```diff
@@ -57,24 +74,45 @@ def create_setting(config, base_con_profile):
     if not setting_ip:
         setting_ip = nmclient.NM.SettingIP6Config.new()
     if config and config.get('enabled'):
-        for address in config.get('address', ()):
-            if iplib.is_ipv6_link_local_addr(address['ip'],
-                                             address['prefix-length']):
-                logging.warning('IPv6 link local address '
-                                '{a[ip]}/{a[prefix-length]} is ignored when '
-                                'applying desired state'.format(a=address))
-            else:
-                naddr = nmclient.NM.IPAddress.new(socket.AF_INET6,
-                                                  address['ip'],
-                                                  address['prefix-length'])
-                setting_ip.add_address(naddr)
-        if setting_ip.props.addresses:
+        is_dhcp = config.get('dhcp', False)
+        is_autoconf = config.get('autoconf', False)
+        if is_dhcp and is_autoconf:
             setting_ip.props.method = (
-                nmclient.NM.SETTING_IP6_CONFIG_METHOD_MANUAL)
-        else:
+                nmclient.NM.SETTING_IP6_CONFIG_METHOD_AUTO)
+            if config.get('address', ()):
+                logging.warning('IPv6 addresses are ignored when DHCP enabled')
+        elif is_dhcp and not is_autoconf:
             setting_ip.props.method = (
-                nmclient.NM.SETTING_IP6_CONFIG_METHOD_LINK_LOCAL)
+                nmclient.NM.SETTING_IP6_CONFIG_METHOD_DHCP)
+            if config.get('address', ()):
+                logging.warning('IPv6 addresses are ignored when DHCP enabled')
+        elif not is_dhcp and is_autoconf:
+            raise NoSupportIpv6DHCPOptionError(
+                'Enable autoconf only is not supported yet')
+        else:
```

Comment:
I already lost track of all these if-else conditions. Could you please separate into functions so it will be easier to read?